draft-ietf-quic-transport.md

```diff
@@ -4697,11 +4697,12 @@ active_connection_id_limit (0x000e):
   to store. This value includes the connection ID received during the handshake,
   that received in the preferred_address transport parameter, and those received
   in NEW_CONNECTION_ID frames.
-  Unless a zero-length connection ID is being used, the value of the
-  active_connection_id_limit parameter MUST be no less than 2. If this
-  transport parameter is absent, a default of 2 is assumed.
-  When a zero-length connection ID is being used, the active_connection_id_limit
-  parameter MUST NOT be sent.
+  The value of the active_connection_id_limit parameter MUST be at least 2.
+  If this transport parameter is absent, a default of 2 is assumed.  If an
+  endpoint uses a zero-length connection ID, the active_connection_id_limit
+  value received from its peer is ignored and not used, as
+  NEW_CONNECTION_ID frames cannot be sent by an endpoint that opts for a
+  zero-length connection ID.
 
```

Comment:
Is the parameter also ignored in future connections where TP is stored?

Reply:
See #3425 for that.

Reply:
Is I wrote in comment to #3425, I'm not sure how 0-length is remembered for future connections when the 0-length value is to be ignored. Does it mean 0-length values are not remembered?

Reply:
If the new connection includes a zero-length connection ID, then the value of the transport parameter is not relevant.  It should still be remembered though.

Reply:
I just found the following in PR #3201 : "When a zero-length connection ID is being used, the active_connection_id_limit  parameter MUST NOT be sent.".

I take it that it is then not possible to remember 0 length?

Reply:
The text "When a zero-length connection ID is being used, the active_connection_id_limit parameter MUST NOT be sent." is removed in this PR.

If 0-length CIDs are used, then this TP is irrelevant. The endpoints still need to remember what value was sent for this TP (or remember the default of 2). Remembering this could be a hardcoded configuration, e.g. if the client always sends 0-length CIDs then it doesn't actually need to remember it.

Reply:
Well, remembering can't be hard-coded as this is a property of the behaviour of a peer and we don't really allow for implementations to only work if a peer uses zero-length connection IDs.  But the central point remains.

Clients don't really have to remember this transport parameter as much as servers unless they intend to send NEW_CONNECTION_ID during 0-RTT.  But putting that in the spec is perhaps a little detailed - we can rely on no one really knowing if they don't remember it.  Just like clients might choose not to remember the limit on unidirectional streams on the basis that they might assume that servers will always provide the small fixed number needed for HTTP to function.